### PR TITLE
kOS toolbar fix

### LINKED ISSUE #951: [bug][able to reproduce] kOS toolbar window appearing in Space Center scene
Steps to reproduce:
* Go to space center scene
* Open any one of the following: Astronaut Center, R&D Center, Mission Control, Admin Building
* Exit back to space center scene
* Open any one of the above buildings again (doesn't have to be same building)
* Exit back to space center scene again
* Toolbar window should now be open despite not having its icon in the toolbar

### PR DESCRIPTION
**Pull request by abenkovskii**
fixes #951
Also fixed this:
screenshot5
P.S. the second bug happens when you first hover over the kOS button.
P.P.S. My IDE changed .csproj, .sln and .userprefs should I revert this changes?

### CODE REVIEW

**Comment by abenkovskii** [comment]
@erendrake Check this out. Can it be added to the next release?

**Comment by ZiwKerman** [comment]
@abenkovskii 

I suggest we bring a more resuable and unified code, we can borrow from here I believe:
https://github.com/MagicSmokeIndustries/InfernalRobotics/blob/develop/InfernalRobotics/InfernalRobotics/Extension/RectExtensions.cs

**Comment by ZiwKerman** [comment]
From what I see in your fix you want to clamp the window to the screen so it is always visible, if it is enabled, and it is what `ClampToScreenEdge(Rect)` does

**Comment by abenkovskii** [comment]
The code you are talking about is not generic enough. It can clamps a rect only to the screen. But kOS needs to clamp it's window to a smaller rect:
screenshot21

**Comment by ZiwKerman** [comment]
What I'm trying to say is we should probably follow the same approach, create some Utility code that will be more reusable, so we can use it not only for Toolbar but for all other windows in kOS

**Comment by abenkovskii** [comment]
I see now. Yes generic approach is probably better.

**Comment by ZiwKerman** [comment]
For example create ClampToRect() and make ClampToScreen() using it

**Comment by ZiwKerman** [comment]
Also it seems your branch is a bit outdated- you should try pulling latest code. We've added kOS Icon to SpaceCenter view some time ago.

**Comment by abenkovskii** [comment]
I know. And I also know that it was added as a "quick and dirty" solution to the first bug this PR solves so I deleted it.

**Comment by abenkovskii** [comment]
[don't merge yet] ok. I'll change it to the more generic variant.

**Comment by abenkovskii** [comment]
[ready] I believe it's ready again

**Comment by ZiwKerman** [comment]
I will try it now
What should I do to test it out?

**Comment by abenkovskii** [comment]
1. Try reproducing this bug: #951
2. Open the VAB, mouseover kOS app launcher button but **don't** click it. kOS window should appear in the right place.
3. Optional: test other situations where kOS toolbar window should appear

**Comment by erendrake** [comment]
I am waiting on @ZiwKerman to test this before i merge. Thank you both for working on this.

File: src/kOS/Screen/KOSToolBarWindow.cs [hunk 1]
```diff
@@ -56,7 +54,7 @@ public class KOSToolBarWindow : MonoBehaviour
         // ReSharper restore RedundantDefaultFieldInitializer
         private Vector2 scrollPos = new Vector2(200,350);
 
-        private static Rect windowRect;
+        private static Rect windowRect; // does anybody know why this is ststic?
```
Comment (ZiwKerman) [inline src/kOS/Screen/KOSToolBarWindow.cs]: Because we only have one window?

**Comment by ZiwKerman** [comment]
Toolbar window now behaves properly, at least I could not reproduce previous bugs.

File: src/kOS/Screen/KOSToolBarWindow.cs (hunk 1 above)
Comment (abenkovskii) [inline src/kOS/Screen/KOSToolBarWindow.cs]: But some per-window fields (e.g. isOpen, clickedOn, width, height) are not static.

**Pull request merged**